front/pages/w/[wId]/builder/assistants/[aId]/index.tsx

```diff
@@ -185,12 +132,91 @@ export const getServerSideProps: GetServerSideProps<{
       };
     }
   }
+
+  return {
+    dataSourceConfigurations,
+    dustAppConfiguration,
+    databaseQueryConfiguration,
+  };
+}
+
+export const getServerSideProps: GetServerSideProps<{
+  user: UserType;
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  plan: PlanType;
+  gaTrackingId: string;
+  dataSources: DataSourceType[];
+  dataSourceConfigurations: Record<string, DataSourceConfig>;
+  dustApps: AppType[];
+  dustAppConfiguration: AssistantBuilderInitialState["dustAppConfiguration"];
+  databaseQueryConfiguration: AssistantBuilderInitialState["databaseQueryConfiguration"];
+  agentConfiguration: AgentConfigurationType;
+  flow: BuilderFlow;
+}> = async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+  const auth = await Authenticator.fromSession(
+    session,
+    context.params?.wId as string
+  );
+
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const subscription = auth.subscription();
+  if (
+    !owner ||
+    !plan ||
+    !user ||
+    !subscription ||
+    !auth.isUser() ||
+    !context.params?.aId
+  ) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const allDataSources = await getDataSources(auth);
+
+  const dataSourceByName = allDataSources.reduce(
```

Comment:
Why not move that in buildInitialState?

Reply:
The data sources are used even when not copying, so it does not fit into `buildInitialState()`.
I think it makes more sense to duplicate it.